lib/dockerregistry/inventory.go

```diff
@@ -1033,6 +1022,22 @@ func (sc *SyncContext) ReadRegistries(
 	sc.ExecRequests(populateRequests, processRequest)
 }
 
+func SplitByKnownRegistries(
+	r RegistryName,
+	rcs []RegistryContext) (RegistryName, ImageName, error) {
+
+	for _, rc := range rcs {
+		//		if inameIdx := strings.LastIndex(string(r), string(rc.Name)); inameIdx != -1 {
+		//			return rc.Name, ImageName(r[inameIdx+1:]), nil
+		//		}
+		if trimmed := strings.TrimPrefix(string(r), string(rc.Name)); len(trimmed) < len(r) {
+			return rc.Name, ImageName(trimmed[1:]), nil
+		}
+	}
+
+	return "", "", fmt.Errorf("oops #2")
```

Comment:
Maybe a better error?  ("unknown registry %q", r) maybe?

Reply:
So sorry about this; I am ashamed to submit such negligent code. ACK